lib/LaTeXML/Util/Pack.pm

```diff
@@ -242,8 +242,9 @@ sub get_archive {
       or (print STDERR "Fatal:I/O:$pathname File $pathname is not readable.");
     my $file_contents = <$FH>;
     close($FH);
-    # Compress all files
-    $archive->addString($file_contents, $file,)->desiredCompressionMethod(COMPRESSION_DEFLATED); }
+    # Compress all files except mimetype
+    my $compression = ($file eq 'mimetype' ? COMPRESSION_STORED : COMPRESSION_DEFLATED);
```

Comment:
I'm sure (or hope :> ) this code could be simpler, but I'm not sure why it's the way it is; maybe @dginev could comment.

But the point is exactly to *not* mangle the mimetype file, or at least, not compress it.

Reply:
> But the point is exactly to _not_ mangle the mimetype file, or at least, not compress it.

Yes, that is definitely correct.

I was just thinking about `if ($format eq 'epub') { $archive->addString('application/epub+zip', 'mimetype', COMPRESSION_STORED); }` before the loop. But indeed maybe there is another reason for this code, e.g. other containers that use the mimetype file in the same way (do they exist?).

Reply:
The (badly named) `get_archive` routine tried to be a single reusable method that could build ZIP archives for output format latexml may need to emit as ZIP. I have only ever used it for building ePubs and for packaging HTML5 for arXiv's builds. I'd say it was only reusable under the assumption no special customization at the ZIP-building stage will be necessary.

The fix Bruce added here corrects a bug I appear to have introduced the last time I patched the sub ( in #1748 ). The bug slipped by as we don't have a test file that would fail on "mimetype" getting compressed, and I wasn't thinking of ePub at the time. Maybe we should add epub _validation_ as part of an extended test suite?

Unrelated, nowadays I'd rename the method to `build_archive`, to add some extra clarity.

If at some point it turns out it is not possible to "simply zip the assets" - which may already be the case with this PR - we could consider a dedicated `build_epub_archive` routine that gets invoked based on `$format`.

Lastly, Pack.pm currently contains some additional auxiliary methods that are not strictly related to "packing" the files, and they can be extracted into new .pm files, if the overall structure is looking too complicated.

Reply:
> Maybe we should add epub _validation_ as part of an extended test suite?

Alternatively, one can make the compressed file reproducible and add the epub itself to the usual tests. I think sorting the file list (in a locale independent way) and setting the modification times to 1970-..., or better yet SOURCE_DATE_EPOCH, via `setLastModFileDateTimeFromUnix` is enough.

Reply:
I suspect `Pack.pm` needs some updating, style modernization etc. Since it's used for several purposes, I'll leave that for now.

Reply:
I've added an `ltx:TOC` for epubs if none present (code is somewhat unmodular, probably want to revisit in the future)